protonvpn-gui: 4.8.2 -> 4.9.5

Steinhagen opened the pull request:
Update the Proton VPN package and dependencies to the latest version

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Steinhagen committed:
python3Packages.proton-vpn-api-core: 0.39.0 -> 0.42.3

Update to the latest package.

Steinhagen committed:
python3Packages.proton-vpn-network-manager: 0.10.2 -> 0.12.13

Update the package to the latest version.

Steinhagen committed:
proton-vpn-local-agent: 1.2.0 -> 1.4.4

Updated the package to the latest stable version.

Steinhagen committed:
protonvpn-gui: 4.8.2 -> 4.9.5

Update the package to the latest version.

Steinhagen commented:
This PR won't work as expected until [this upstream issue](https://github.com/ProtonVPN/python-proton-vpn-api-core/issues/6) gets resolved. Because of it, the proton-vpn-api-core will try to generate a number of directories on a simple import.

SebTM commented:
> This PR won't work as expected until [this upstream issue](https://github.com/ProtonVPN/python-proton-vpn-api-core/issues/6) gets resolved. Because of it, the proton-vpn-api-core will try to generate a number of directories on a simple import.

Can we patch this until it's solved upstream?

JaysFreaky commented:
Is there any chance to bump this to 4.9.4?

Apparently, 4.9.4 resolves an issue where the VPN won't connect if IPv6 is disabled.
https://github.com/ProtonVPN/proton-vpn-gtk-app/issues/57#issuecomment-2707686822

nyabinary commented:
Why is this still in draft btw :P, wireguard doesn't work in the ver packaged in nixpkgs

Steinhagen commented:
I'm not exactly sure how to fix this for Nix right now. Any help with this would be appreciated.

nyabinary commented:
> I'm not exactly sure how to fix this for Nix right now. Any help with this would be appreciated.

Maybe ask in the Matrix nixpkgs chat, or the forums?

Steinhagen commented:
I had a discussion on the Matrix chat and the idea was that this should be fixed in the upstream repository.
That being said, I've left additional comments describing the issues and will follow further with them to see how this can be resolved.

Artturin committed:
python312Packages.{proton-vpn-api-core,proton-vpn-network-manager}: Fix import tests

When pyxdg is available the dirs will use xdg vars

https://www.github.com/ProtonVPN/python-proton-core/blob/e87d1e3b5fc70883adc02ef2e2d9b0eb360b83ba/proton/utils/environment.py#L56-L102

Artturin commented:
Build fixed and the gui launches, sadly wireguard still doesn't work for me.

Steinhagen commented on pkgs/applications/networking/protonvpn-gui/default.nix:
```diff
@@ -37,12 +38,16 @@ buildPythonApplication rec {
-  ];
+  buildInputs =
+    [
+      libnotify # gir
```
Maybe I'm missing something, but what do you mean by the gir comment?

Artturin replied:
Changed to the `gir typelib is used`

Steinhagen commented:
> Build fixed and the gui launches, sadly wireguard still doesn't work for me (it connects and `wg show` has it but can't access websites).

What have you selected in the 'Net Shield' option inside the application settings? Can you try with 'Off'?

Artturin committed:
protonvpn-gui: Add missing dep

https://www.github.com/ProtonVPN/proton-vpn-gtk-app/blob/1d44f9aa7e4239685c1759bf9c106b3a14708e76/proton/vpn/app/gtk/__init__.py#L31

Artturin commented:
> > Build fixed and the gui launches, sadly wireguard still doesn't work for me (it connects and `wg show` has it but can't access websites).
> 
> What have you selected in the 'Net Shield' option inside the application settings? Can you try with 'Off'?

Doesn't work with it on or off

Steinhagen commented:
Gave this update a try and everything worked as expected. Regarding the Wireguard settings on my side, I have Kill Switch Enabled (Standard) and Net Shield Off.

Artturin commented:
> Gave this update a try and everything worked as expected. Regarding the Wireguard settings on my side, I have Kill Switch Enabled (Standard) and Net Shield Off.

With the kill switch it works :sweat_smile: I hadn't tried that one. Though no ipv6.

nyabinary commented:
> > Gave this update a try and everything worked as expected. Regarding the Wireguard settings on my side, I have Kill Switch Enabled (Standard) and Net Shield Off.
> 
> With the kill switch it works 😅 I hadn't tried that one yet for some reason. Though no ipv6.

I mean it still should work without kill switch on right?

Steinhagen commented:
I think it should, but I haven't looked deeper into this. Probably a ticket should be created for this, or someone should try this behavior on a non-NixOS distro to see if it's an upstream issue. If it is, a bug should be raised for this.
Same would be ideal to test the ipv6 connectivity using Wireguard. If everything works on those distros we should have at least a bug tracking this issue in nixpkgs.

Artturin commented:
With https://github.com/NixOS/nixpkgs/issues/307462#issuecomment-2750133149 it works without the kill switch
